sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -207,11 +229,31 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     blobOptions.AccessConditions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
     blobOptions.AccessConditions.LeaseId = options.AccessConditions.LeaseId;
     auto result = m_blobContainerClient.Delete(blobOptions);
-    Models::FileSystemDeleteResult ret;
-    return Azure::Core::Response<Models::FileSystemDeleteResult>(
+    Models::DeleteFileSystemResult ret;
+    ret.Deleted = true;
+    return Azure::Core::Response<Models::DeleteFileSystemResult>(
         std::move(ret), result.ExtractRawResponse());
   }
 
+  Azure::Core::Response<Models::DeleteFileSystemResult> FileSystemClient::DeleteIfExists(
+      const DeleteFileSystemOptions& options) const
+  {
+    try
+    {
+      return Delete(options);
+    }
+    catch (StorageException& e)
+    {
+      if (e.ErrorCode == Details::ContainerNotFound)
+      {
+        Models::DeleteFileSystemResult ret;
+        ret.Deleted = false;
+        return Azure::Core::Response<Models::DeleteFileSystemResult>(ret, std::move(e.RawResponse));
+      }
+      throw;
+    }
+  }
```

Comment:
why not call `BlobContainerClient::DeleteIfExists`?

Reply:
It does not have `Deleted` in the response.